t/unit/backends/test_elasticsearch.py

```diff
@@ -80,3 +80,33 @@ def test_backend_params_by_url(self):
             assert x.scheme == 'elasticsearch'
             assert x.host == 'localhost'
             assert x.port == 9200
+
+    def test_index(self):
+        x = ElasticsearchBackend(app=self.app)
+        x.doc_type = 'test-doc-type'
+        x._server = Mock()
+        x._server.index = Mock()
+        expected_result = dict(
+            _id=sentinel.task_id,
+            _source={'result': sentinel.result}
+        )
+        x._server.index.return_value = expected_result
+
+        body = {"field1": "value1"}
+        x._index(id=sentinel.task_id, body=body, kwarg1='test1')
+        x._server.index.assert_called_once_with(
+            id=sentinel.task_id,
```

Comment:
I think you should convert this object to string in order to take into account this [change](https://github.com/celery/celery/pull/3961/files#diff-731afbc33e1f1197b570c2d5506600d2R109), so that the assertion works:

```
x._server.index.assert_called_once_with(
    id=string(sentinel.task_id),
```